pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -40,25 +40,42 @@ import (
 )
 
 const (
-	socketName       = "/tmp/device_plugin/server.sock"
-	pluginSocketName = "/tmp/device_plugin/device-plugin.sock"
 	testResourceName = "fake-domain/resource"
 )
 
+func tmpSocketDir() (socketDir, socketName, pluginSocketName string, err error) {
+	socketDir, err = ioutil.TempDir("", "device_plugin")
+	if err != nil {
+		return
+	}
+	socketName = socketDir + "/server.sock"
+	pluginSocketName = socketDir + "/device-plugin.sock"
+	return
+}
+
 func TestNewManagerImpl(t *testing.T) {
-	_, err := newManagerImpl(socketName)
+	socketDir, socketName, _, err := tmpSocketDir()
```

Comment:
Wondering if we should have a [TestMain](http://cs-guy.com/blog/2015/01/test-main/) function here.
That would prevent writing this setup and teardown code each time.

What do you think?

Reply:
The problem in issue #59488 seemed to be that different tests in the same test program share the same socket dir and somehow interfere with each other. I am all ears if someone can explain why that happens, because it has not become clear to me.

If we use TestMain, we end up in the same situation and issue #59488 will probably occur again.

If it wasn't for that I'd agree with you, TestMain would be more suitable than the per-test boiler plate code.